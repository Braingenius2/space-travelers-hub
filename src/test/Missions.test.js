import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'; // For better DOM assertions
import Missions from '../components/missions';

// Mocking useDispatch and useSelector
jest.mock('react-redux', () => ({
  useDispatch: () => jest.fn(),
  useSelector: (selectorFn) => selectorFn(
    { missions: { missions: [] } },
  ),
}));

test('renders the Empty Missions component correctly', () => {
  const { container } = render(<Missions />);
  expect(container).toMatchSnapshot();
});
