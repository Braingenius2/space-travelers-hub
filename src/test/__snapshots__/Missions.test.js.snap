// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Empty Missions component correctly 1`] = `
<div>
  <div
    style="display: flex; flex-direction: column; align-items: center;"
  >
    <table
      class="table table-striped table-bordered table-hover"
      style="width: 90%;"
    >
      <thead>
        <tr>
          <th>
            Mission
          </th>
          <th>
            Descripton
          </th>
          <th>
            Status
          </th>
          <th>
            <p
              style="display: none;"
            >
              Action
            </p>
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
